src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -271,6 +276,42 @@ private static ImmutableArray<Diagnostic> EmitRegexMethod(IndentedTextWriter wri
                 writer.WriteLine($"            }}");
             }
 
+            if ((requiredHelpers & RequiredHelperFunctions.IsBoundary) != 0)
+            {
+                writer.WriteLine();
+                writer.WriteLine($"            /// <summary>Determines whether the character at the specified index is a boundary.</summary>");
+                writer.WriteLine($"            [global::System.Runtime.CompilerServices.MethodImpl(global::System.Runtime.CompilerServices.MethodImplOptions.AggressiveInlining)]");
+                writer.WriteLine($"            private static bool IsBoundary(global::System.ReadOnlySpan<char> inputSpan, int index)");
+                writer.WriteLine($"            {{");
+                writer.WriteLine($"                const char ZeroWidthNonJoiner = '\\u200C', ZeroWidthJoiner = '\\u200D';");
+                writer.WriteLine();
+                writer.WriteLine($"                return (index > 0 && IsBoundaryWordChar(inputSpan[index - 1])) !=");
```

Comment:
I think we could avoid a comparison / bounds check if we instead wrote it like this:
```suggestion
                writer.WriteLine($"                int indexM1 = index - 1;");
                writer.WriteLine($"                return ((uint)indexM1 < (uint)inputSpan.Length && IsBoundaryWordChar(inputSpan[indexM1])) !=");
```
Same for IsECMABoundary below.

[SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNYgGYmpBkIYBvGgzMMxUXguwYYTckgbAIESQwCSuAEIRuAE2woAE9yAAo2GGx/AHkuSWDZMWwuAB4wAAsggD4GXi4xDgwklLQ8nnL/GAQASlNzE2pzZqYAdnqWlrD8qoQGXMYAMkHPHz8uQJCAdWh/ISyoboKikq4AbR7qhjgGcgBdGpqGAEIAXg7O8zCwjnyMGs2+1IYbu4fl4uSuFgAZGC4AOYYDIMYajXwBILBGZQOYLJaFT4pDYTaoHGoaJrmAC+FwulmstnszCcLjc4PGk2CpAiUVi8USX3SC1y+URqzKd0q1TqWLMjUu5Qw3IQAFlyAxTiLtrtMYLiO0+YKXq8eO9euKGM9Vfc2Ssvr9/kCQWCvBCJlCYXCggj9cjHuL0SdzkrBddbmrHlqXh7dR9VobAcDQSMzZTLbN5ja9Uj1o90XKcXilWtRTBgRB/B5xJIwmmM1mczExHwIFxcCwAHIQDzxfL5AE1Pb4qw2OwOUmudxhyHTSPwzJBBiZI6nXIAM2wklwMEx2KAA==)